lib/plug/parsers.ex

```diff
@@ -236,4 +241,9 @@ defmodule Plug.Parsers do
       raise UnsupportedMediaTypeError, media_type: "#{type}/#{subtype}"
     end
   end
+
+  defp merge_path_params(%{path_params: %Plug.Conn.Unfetched{}} = conn), do: conn
```

Comment:
That's right, the original params have to be preserved (there are tests to make sure of this anyway).

@josevalim: in the current Parsers code, it looks like `query_params` will always be fetched. But there are conditions in which `body_params` will still be unfetched after calling `Plug.Parsers.call/2` . For, example, in [this test](https://github.com/elixir-lang/plug/blob/master/test/plug/parsers_test.exs#L112).

However, in your example code above, `body_params` will now never be unfetched. It doesn't seem like it would be a problem, but I thought I should point it out, since technically it's changing the behavior.